### DISCUSSION TITLE
[bug] Issue templates "blank_issues_enabled: false" not enforced

### DISCUSSION THREAD

**Discussion by cytopia** [community/community#14287] [Projects and Issues]
> **Note:** Originally created [there](https://github.community/t/bug-issue-templates-blank-issues-enabled-false-not-enforced/242312), but asked to duplicated here.

### Description
When configuring issue templates and a [template chooser](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) with `blank_issues_enabled: false` is set (which essentially means that you an only create issues by the defined templates), I can still create plain issues, which should not be allowed.

In [my repository](https://github.com/cytopia/devilbox) I have created the following issue templates (click the image to navigate there):

[![](https://aws1.discourse-cdn.com/github/original/3X/1/c/1cdc88de54407f76bd3e5d744d284be4c882eda2.png)](https://github.com/cytopia/devilbox/issues/new/choose)

I have enforced usage of templates via `.github/ISSUE_TEMPLATE/config.yml`:
```yaml
---
blank_issues_enabled: false
...
```

### Steps to reproduce

1. Click on **create new issue**: [https://github.com/cytopia/devilbox/issues/new/choose](https://github.com/cytopia/devilbox/issues/new/choose)
2. Select **feature request** [https://github.com/cytopia/devilbox/issues/new?assignees=&labels=feature&template=feature_request.yml&title=%5BFeature%5D%3A+](https://github.com/cytopia/devilbox/issues/new?assignees=&labels=feature&template=feature_request.yml&title=%5BFeature%5D%3A+)

I can then simply remove the URL GET parameter and keep the URL as such:

[https://github.com/cytopia/devilbox/issues/new](https://github.com/cytopia/devilbox/issues/new) 

This will then allow me to create a blank issue.

### Additional issues

Besides this, I can also alter the above URL GET params to specify different labels or other assignees, which should not be possible.

**Comment by brillout** [comment]
It works for this repository https://github.com/cytopia/devilbox/issues/new (notice the URL redirect) but not for this one https://github.com/brillout/vite-plugin-ssr/issues/new (there is no redirect).

Although both have `blank_issues_enabled: false`:
- devilbox: https://github.com/cytopia/devilbox/blob/6a045274569c54e02e333aaeb18104f52bb47a3f/.github/ISSUE_TEMPLATE/config.yml#L2
- vite-plugin-ssr: https://github.com/brillout/vite-plugin-ssr/blob/8ad0e41e7876bf74f35ebd5186b0453715e8457d/.github/ISSUE_TEMPLATE/config.yml#L1

Why does it work for the former but not the latter?

**Reply by cytopia** [reply]
> It works for this repository https://github.com/cytopia/devilbox/issues/new (notice the URL redirect) but not for this one https://github.com/brillout/vite-plugin-ssr/issues/new (there is no redirect).

@[brillout](https://github.com/brillout) For me it is exactly the other way round. I get a redirect on your issue creation and you get one on mine. Could it be that it will only not work for the owner of the repository?

**Reply by brillout** [reply]
Seems like it yes. Did you receive any new plain non-template issue? If not, that would explain why.

**Reply by cytopia** [reply]
No, only received valid templated issues since enabled.

**Reply by sapegin** [reply]
Hm, I did get a few issues from non-collaborators in various repositories ([here](https://github.com/sapegin/textlint-rule-terminology), for example), even with `blank_issues_enabled: false`, and to me it looks like a bug.

**Comment by GMNGeoffrey** [comment]
I agree that something is weird here, but I don't think it's obvious that `blank_issues_enabled: false` in the `ISSUE_TEMPLATES/config.yml` should imply that it's impossible to create blank issues even from a direct link.

The [docs for issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) say you're "encouraging" contributors to use the templates. In our project, we've actually tried to make the link for blank issues more prominent because getting templated issues from regular collaborators results in a ton of annoying boilerplate: https://github.com/iree-org/iree/pull/9860. I did this by trying to create a "blank" template, but it turns out that it's impossible to have a template that doesn't insert extra text into the issue (boo), so we switched to making the blank issue a link to create a blank issue: https://github.com/iree-org/iree/pull/10043. To keep the template picker clean, we set `blank_issues_enabled: false`. This was a decision purely motivated by cosmetics. Then we just discovered that apparently this makes it impossible to create a blank issue even via direct link if you're not a collaborator: https://github.com/iree-org/iree/pull/10197. I don't know how we would have discovered this behavior, since it's not documented AFAICT and the people doing this are obviously all collaborators.

So regardless, the behavior should all be properly documented. It seems really weird that it's silently different for collaborators and non-collaborators.

Having `blank_issues_enabled: false` imply no creation of regular blank issues would be fine with me if it were documented and I could create my own issue template that was *actually* blank. The [behavior](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#attributes-1) that a textarea in an issue template always prints its label in the issue body seems bad regardless.

**Comment by Arlodotexe** [comment]
It looks like behavior was recently changed so that https://github.com/MyOrg/MyRepo/issues/new/ always redirects to https://github.com/MyOrg/MyRepo/issues/new/choose. 

I'm not a fan of this change because while the issue templates are always for contributors, they're not always for maintainers. I have high volumes of tickets to file and organize as a maintainer, being restricted to the templates used to guide new and unfamiliar contributors is more hindrance than help.

**Reply by noah-nuebling** [reply]
I relied on the old behavior to let my 'Feedback Assistant' website ([link](https://noah-nuebling.github.io/mac-mouse-fix-feedback-assistant/?type=bug-report)) create GitHub Issues, while leaving `blank_issues_enabled: false`

Now I had to set `blank_issues_enabled: true`, which causes a 'Blank issue' option to show up prominently at the top of the `github.com/.../issues/new/choose` page. 

Is there a way to prevent this, or at least get the 'Blank issue' option to show up at the bottom of the list instead of the top?

**Reply by flying-sheep** [reply]
the “open blank issue” option used to be an unobtrusive link in the small text line below the big buttons which are created from the templates and contact links.

now it’s one of the big buttons and displayed even above the contact links, which makes it far too encouraging to click for users who don‘t want to think.

**Reply by jviall** [reply]
I'm having the complete opposite experience, where for a particular repo navigating to `/issues/new/choose` is redirecting to `/issues/new`, obfuscating the picker.

**Comment by vvolkgang** [comment]
Blank issues are still possible:
1. Via API - Apps like OctoDroid allow users to create issues without handling templates
2. Via Browser - just set a non existing template https://github.com/cytopia/devilbox/issues/new?template=nope

With the advent of tools using the API directly we're seeing an uptick in issues created without templates, breaking automations.
